Implements ROUGE score with unit tests and documentation

rohithmsr opened the pull request:
## Details

Introduces a new class ROUGE to implement ROUGE heuristic metric using the `rouge-score` [library](https://pypi.org/project/rouge-score/)

## Issues

Resolves https://github.com/comet-ml/opik/issues/979

## Testing

- Includes unit tests covering various argument types and error scenarios.
- Ensures compliance with linting and code quality standards.

## Documentation

Adds documentation for ROUGE in `fern/docs/evaluation/metrics/heuristic_metrics.mdx` including an introduction, dependencies, usage examples, and references.

## Considerations

I am still gaining familiarity with pytest and the ROUGE metric. Feedback and suggestions are welcome! I referred to @kadamrahul18's PR https://github.com/comet-ml/opik/pull/1006 as a reference

alexkuzmik commented on sdks/python/src/opik/evaluation/metrics/heuristics/rouge.py:
```diff
@@ -0,0 +1,129 @@
+        rouge_type: str = "rouge1",
+        use_stemmer: bool = False,
+        split_summaries: bool = False,
+        tokenizer: Any | None = None,
```
`Any | None` -> `Optional[Any]`
`|` typing operator has appeared in python 3.10 only and we support python3.8+

rohithmsr replied:
Updated the type

rohithmsr replied:
@alexkuzmik , Is this the reason why some checks were failing?

alexkuzmik reviewed (changes requested):
Thanks @rohithmsr, the PR looks good!
I left 2 small comments to fix.

rohithmsr commented on sdks/python/src/opik/evaluation/metrics/heuristics/rouge.py:
```diff
@@ -0,0 +1,129 @@
+from opik.evaluation.metrics import base_metric, score_result
+
+try:
+    from rouge_score import rouge_scorer
```
Updated as requested

rohithmsr commented:
Hi @alexkuzmik 
I've addressed the comments, please re-look when you have bandwidth. Let me know if anything is required from my end.
Thanks!

cc: @jverre

alexkuzmik commented:
@rohithmsr I merged your PR. There was one small issue related to imports, which I fixed separately. 
We're planning to clean up the CI a bit so that there are no always failing integration tests for external contributors, so it will be easier to see if unit tests are failing.

FYI https://github.com/comet-ml/opik/pull/1717
